src/dreyfus_index_updater.erl

```diff
@@ -31,6 +31,14 @@ update(IndexPid, Index) ->
     erlang:put(io_priority, {view_update, DbName, IndexName}),
     {ok, Db} = couch_db:open_int(DbName, []),
     try
+        {ok, IdxPurgeSeq} = clouseau_rpc:get_purge_seq(IndexPid),
+        FoldFun = fun(PurgeSeq, {Id, _Revs}, Acc) ->
+            clouseau_rpc:delete(IndexPid, Id),
```

Comment:
Hm, I'm still not convinced. If we look at couch_index/couch_mrview that seems to do the right thing. It purges the items and then updates the index, before returning the updated mrview state to the caller. Here we will make the intermediate state, where the Documents in the Lucene index are removed, visible to search results.